### PR TITLE
Dashboard Visual Style Revamp (2020Mar)

### LINKED ISSUE #862: [Demand] Dashboard Visual Style Revamp (2020Mar)
## Metadata

Field | Value
--- | ---
Previous Stage | ...
Figma Link | https://www.figma.com/file/4DmpRxZk4jDeqrfA5xrPCp/Maskbook-Dashboard?node-id=1%3A168
Expected UX prototype due | 2020-W??-5
Actual UX prototype release | -
Expected engineering due | 2020-W??-5
Corresponding Pull Request | -
Completion Version | -

## Catalogue

- Home
- Personas
- Wallets

## Notes

- This is a draft. More contents will be added when Figma prototypes are ready for review.

### PR DESCRIPTION
**Pull request by SunriseFox**
Close #862

### CODE REVIEW

File: src/utils/theme.ts [hunk 1]
```diff
@@ -29,6 +36,20 @@ const baseTheme = (theme: 'dark' | 'light') =>
         breakpoints: {
             values: { xs: 0, sm: 600, md: 1024, lg: 1280, xl: 1920 },
         },
+        overrides: {
+            MuiButton: {
+                root: {
+                    textTransform: 'none',
```
Comment (Jack-Works) [inline src/utils/theme.ts]: Not so material🤔

File: src/utils/theme.ts [hunk 2]
```diff
@@ -29,6 +36,20 @@ const baseTheme = (theme: 'dark' | 'light') =>
...
+                    minWidth: '100px',
+                },
+            },
+        },
+        props: {
+            MuiButton: {
+                disableElevation: true,
```
Comment (Jack-Works) [inline src/utils/theme.ts]: Not so material 🤔🤔

File: src/extension/options-page/index.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        </I18nextProvider>
+    )
+}
+
+const useStyles = makeStyles(theme =>
+    createStyles({
+        wrapper: {
+            '--monospace': 'SFMono-Regular,Consolas,Liberation Mono,Menlo,monospace',
+
+            '--primary': theme.palette.primary.main,
```
Comment (Jack-Works) [inline src/extension/options-page/index.tsx]: strange... why do you want to use css variable when the value is directly inherit from the theme?

File: src/extension/options-page/index.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            '--drawerBodySelected': theme.palette.type === 'dark' ? '#114097' : 'var(--textOnPrimary)',
+            '--drawerTextSelected': theme.palette.type === 'dark' ? 'var(--textOnPrimary)' : 'var(--primary)',
+            '--listSelectedIndicator': 'var(--primary)',
+
+            position: 'absolute',
+            width: '100vw',
+            height: '100vh',
+            backgroundColor: 'var(--background)',
+            // background: 'var(--background) bottom 10px right 10px no-repeat url(/maskface.svg)',
+            display: 'grid',
```
Comment (Jack-Works) [inline src/extension/options-page/index.tsx]: 🌟👍

File: src/extension/options-page/index.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+function Dashboard() {
+    const { t } = useI18N()
+    const classes = useStyles()
+
+    const routers: [string, string, JSX.Element][] = [
+        ['Personas', '/personas/', <PeopleOutlinedIcon />],
+        /* without redpacket */
+        ...(webpackEnv.target === 'WKWebview'
+            ? []
+            : ([['Wallets', '/wallets/', <CreditCardIcon />]] as [string, string, JSX.Element][])),
```
Comment (Jack-Works) [inline src/extension/options-page/index.tsx]: use `as const` instead?

File: package.json [hunk 6]
```diff
@@ -87,6 +87,7 @@
     "qrcode": "^1.4.4",
     "react": "^16.13.0",
     "react-dom": "^16.13.0",
+    "react-feather": "^2.0.3",
```
Comment (Jack-Works) [inline package.json]: 🤔 (Encourage to use icons in `@material-ui/icons` to create a more consistent vision...)

File: src/extension/options-page/DashboardComponents/AbstractTab.tsx [hunk 7]
```diff
@@ -1,50 +1,33 @@
...
 
-    return (
-        <Typography role="tabpanel" hidden={value !== index} {...{ component: 'div' }} {...other}>
-            <Box height={height} p={p ?? 3}>
-                {children}
-            </Box>
-        </Typography>
-    )
+const TabPanel = (props: TabPanelProps & { value: number; index: number }) => {
+    const { value, index, display, label, ...other } = props
```
Comment (Jack-Works) [inline src/extension/options-page/DashboardComponents/AbstractTab.tsx]: label seems like unused

Comment (SunriseFox) [reply src/extension/options-page/DashboardComponents/AbstractTab.tsx]: things may still be adjusted... they are not very stable... don't be in a hurry

File: src/extension/options-page/DashboardComponents/ProviderLine.tsx [hunk 8]
```diff
@@ -1,56 +1,73 @@
...
+        },
     }),
 )
 
 interface ProviderLineProps {
     network: string
     connected?: boolean
     userId?: string
-    onConnect?: () => void
+    onAction?: () => void
```
Comment (Jack-Works) [inline src/extension/options-page/DashboardComponents/ProviderLine.tsx]: Isn't this name too ambigious?

File: src/extension/options-page/DashboardComponents/SpacedButtonGroup.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import { Box, makeStyles, createStyles, BoxProps } from '@material-ui/core'
+import classNames from 'classnames'
+
+const useStyles = makeStyles((theme) =>
+    createStyles({
+        buttonGroup: {
+            flexGrow: 0,
+            flexShrink: 0,
+            '& > *:not(:last-child)': {
+                marginRight: theme.spacing(1),
```
[the remark this answers is not in the document]
Comment (SunriseFox) [reply src/extension/options-page/DashboardComponents/SpacedButtonGroup.tsx:11]: I don't think it's as *simple* as this...

File: src/extension/options-page/DashboardComponents/FooterLine.tsx [hunk 10]
```diff
@@ -23,14 +24,18 @@ const useStyles = makeStyles((theme: Theme) =>
     }),
 )
 
-const FooterLink = function (props: React.PropsWithChildren<{ href: string; title?: string } | { to: string }>) {
+const FooterLink = function (
+    props: React.PropsWithChildren<{ href?: string; title?: string; to?: string; onClick?(): void }>,
```
[the remark this answers is not in the document]
Comment (SunriseFox) [reply src/extension/options-page/DashboardComponents/FooterLine.tsx:28]:
```suggestion
    type FooterLinkBaseProps = { title?: string }
    type FooterLinkLinkProps = FooterLinkBaseProps & { to: string }
    type FooterLinkAnchorProps = FooterLinkBaseProps & { href: string }
    type FooterLinkAnchorButtonProps = FooterLinkBaseProps & { onClick(e: React.MouseEvent<HTMLAnchorElement>): void }

    type FooterLinkProps = FooterLinkLinkProps | FooterLinkAnchorProps | FooterLinkAnchorButtonProps
```

...

File: src/utils/theme.ts [hunk 11]
```diff
@@ -29,6 +36,83 @@ const baseTheme = (theme: 'dark' | 'light') =>
...
+                              border: '1px solid rgba(255, 255, 255, 0.6)',
+                              color: '#FFF',
+                              '&:hover': {
+                                  border: '1px solid rgba(255, 255, 255, 1)',
+                              },
+                          }
+                        : {}),
+                },
+                textPrimary: {
+                    ...(theme === 'dark'
```
[the remark this answers is not in the document]
Comment (SunriseFox) [reply src/utils/theme.ts:57]: I think maybe a patch / update library is better...

[the remark this answers is not in the document]
Comment (Jack-Works) [reply src/utils/theme.ts:57]: yes immer is that kind of library 🎉

File: src/extension/options-page/Dialog/Database.tsx [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        <DashboardDialogCore {...props}>
+            <DashboardDialogWrapper
+                size="small"
+                icon={<DatabaseIcon />}
+                iconColor="#699CF7"
+                primary={t('restore_database')}
+                secondary={
+                    requiredPermissions ? t('dashboard_permission_required') : t('dashboard_restore_database_hint')
+                }>
+                <input type="file" accept="application/json" ref={ref} onChange={fileReceiver} hidden />
```
[the remark this answers is not in the document]
Comment (SunriseFox) [reply src/extension/options-page/Dialog/Database.tsx:89]: Adopted

File: src/utils/theme.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (SunriseFox) [reply src/utils/theme.ts:57]: I don't think it is suitable for this... :awsl:

File: src/extension/options-page/Dialog/Base.tsx [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+)
+
+const dialogTheme = (theme: Theme): Theme => ({
+    ...theme,
+    overrides: {
+        ...theme.overrides,
+        MuiOutlinedInput: {
+            ...theme.overrides?.MuiOutlinedInput,
+            notchedOutline: {
+                ...theme.overrides?.MuiOutlinedInput?.notchedOutline,
```
[the remark this answers is not in the document]
Comment (SunriseFox) [reply src/extension/options-page/Dialog/Base.tsx]: changed to _.merge

File: src/utils/theme.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (SunriseFox) [reply src/utils/theme.ts]: changed to _.merge

File: src/extension/options-page/DashboardComponents/ActionButton.tsx [hunk 14]
```diff
@@ -1,12 +1,75 @@
...
+        state: ThrottledButtonState.Normal,
+    })
+    const classes = useDangerStyles()
+    const hookedClick = useCallback(
+        (e) => {
+            e.stopPropagation()
+            if (loading.state !== ThrottledButtonState.Normal) return
+            setLoading({ state: ThrottledButtonState.Clicked })
+            const timer: number = window.setTimeout(
+                () => setLoading({ state: ThrottledButtonState.Loading, timer }),
```
[the remark this answers is not in the document]
Comment (SunriseFox) [reply src/extension/options-page/DashboardComponents/ActionButton.tsx:51]: seems hard to integrate with the button...

File: src/extension/options-page/DashboardComponents/ProviderLine.tsx (hunk 8 above)
Comment (SunriseFox) [reply src/extension/options-page/DashboardComponents/ProviderLine.tsx:43]: maybe an alias of `onClick`... or `onTrigger`

File: src/extension/options-page/Dialog/Database.tsx [hunk 15]
```diff
@@ -39,29 +40,38 @@ export function DashboardDatabaseRestoreDialog(props: WrappedDialogProps) {
...
+                        enqueueSnackbar(t('done'), { variant: 'success' })
+                        props.onClose()
+                    } else {
+                        setJson(json)
+                        setRequiredPermissions(permissions)
+                    }
+                } catch (e) {
+                    enqueueSnackbar(`Error: ${e.message}`, { variant: 'error' })
+                } finally {
+                    resolve()
```
Comment (Jack-Works) [inline src/extension/options-page/Dialog/Database.tsx:66]: This promise seems isn't get used, so why it needs to resolve?

Comment (SunriseFox) [reply src/extension/options-page/Dialog/Database.tsx:66]: so the button can track this promise and wait for it... (and maybe for the dialog to track it in the future...

File: src/extension/options-page/Router/Wallets.tsx [hunk 16]
```diff
@@ -38,7 +38,7 @@ const useStyles = makeStyles((theme) =>
             height: '100%',
             flex: '0 0 auto',
             borderRight: `1px solid ${theme.palette.divider}`,
-            overflow: 'auto',
+            overflowY: 'auto',
```
Comment (SunriseFox) [inline src/extension/options-page/Router/Wallets.tsx:41]: I think `overflowY: 'auto'` will set overflowX: 'auto' automatically...

File: src/components/shared-settings/settings.ts [hunk 17]
```diff
@@ -3,6 +3,14 @@ import type { ValueRef } from '@holoflows/kit/es'
 import { MessageCenter } from '../../utils/messages'
 import i18nNextInstance, { i18n } from '../../utils/i18n-next'
 
+/**
+ * Does the night mode on
+ */
+export const nightModeSetting = createNewSettings<boolean>('nightMode', false, {
```
Comment (Jack-Works) [inline src/components/shared-settings/settings.ts:9]: The default value should be a media query to query the user default.

**Pull request closed without merging**